kuma/documenturls/models.py

```diff
@@ -0,0 +1,90 @@
+import requests
+
+from django.db import models
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from redo import retrying
+
+
+def download_url(url, retry_options=None):
+    retry_options = retry_options or {
+        "retry_exceptions": (requests.exceptions.Timeout,),
+        "sleeptime": 2,
+        "attempts": 5,
+    }
+    with retrying(requests.get, **retry_options) as retrying_get:
+        response = retrying_get(url, allow_redirects=False)
+        response.raise_for_status()
+
+        return response
+
+
+class DocumentURL(models.Model):
+    """There are documents we look up for things like bookmarks and notes.
+    These are not legacy Wiki documents but rather remote URLs.
+    """
+
+    # E.g. /en-us/docs/web/javascript (note that it's lowercased!)
+    # (the longest URI in all our of content as of mid-2021 is 121 characters)
+    uri = models.CharField(max_length=250, unique=True, verbose_name="URI")
+    # E.g. https://developer.allizom.org/en-US/docs/Web/JavaScript
+    absolute_url = models.URLField(verbose_name="Absolute URL")
```

Comment:
The `absolute_url` field is always constructed from the incoming URI right? It seems to me that we don't need to save it. Can't we just construct it from the `uri`  (e.g. `f"{settings.BOOKMARKS_BASE_URL}{uri}/index.json"`) whenever we need it? I realize that the `uri` is always lower-case, but that shouldn't matter when using the `absolute_url` for fetching.

Reply:
I'm torn. The intention of this field is more as an "audit trail". 
The field is for: **What URL was used (and should continue to be used) for download the metadata?**
The code comment there was wrong, so I changed it. 
It's basically the URL we're sending to `requests.get(HERE)` when we fetch the metadata. 
Yes, its value *can* be reconstructed based on `settings.BOOKMARKS_BASE_URL` and `uri`, but there's some sort of comfort to know that we have a record of which URL was used to `requests.get()` in case we ever change `settings.BOOKMARKS_BASE_URL` or what if it's related to Deep-dives, then you can't so easily "reverse" from the `uri` value. Perhaps the `/api/v1` endpoint has changed for fetching deep-dives' metadata. 

Perhaps, before I jump into deleting it, you and I sleep on it a smidge longer.